bundles/mapping/mapwfs2/service/StatusHandler.js

```diff
@@ -1,118 +1,170 @@
+      });
+      var sb = this.sandbox;
+      var loadEvent = sb.getEventBuilder('WFSStatusChangedEvent')(data.layerId);
+      loadEvent.setRequestType(loadEvent.type.image);
 
-            var sb = this.sandbox;
-            var loadEvent = sb.getEventBuilder('WFSStatusChangedEvent')(error.layerId);
-            loadEvent.setStatus(loadEvent.status.error);
-            loadEvent.setRequestType(loadEvent.type.image);
-            sb.notifyAll(loadEvent);
-       	}
+      if(data.success) {
+        this.__log('Back to normal for layer:', data.layerId, status.error);
+        status.error = [];
+        loadEvent.setStatus(loadEvent.status.complete);
+        // No operations for wfs layer in case of true, e.g. manual refresh / feature data refresh
+        loadEvent.setNop(data.success_nop);
+      }
+      else {
+        status.error.push('error');
+        loadEvent.setStatus(loadEvent.status.error);
+      }
+      sb.notifyAll(loadEvent);
+      this.__log('WFS complete', data);
+    },
+    handleChannelStatus : function(data) {
+      var me = this;
+      // {"layerId":"5","message":"started","reqId":9}
+      // {"layerId":"15","success":true,"message":"completed","reqId":3}
+      if(data.message === 'started') {
+        this.__log('Processing started for layer ' + data.layerId + ' (req: ' + data.reqId + ')');
+        if(typeof this.timer === 'undefined'){
+          this.timer = setTimeout(function(){
+            me.__log('Processing layer ' + data.layerId + ' takes longer than excpected');
+          }, 4000);
+        }
+        return;
+      }
+      else if(data.message === 'completed') {
+        clearTimeout(this.timer);
+        if(data.success) {
+          this.__log('Processing finished for layer ' + data.layerId + ' (req: ' + data.reqId + ')');
+        }
+        else {
+          this.__log('Processing error for layer ' + data.layerId + ' (req: ' + data.reqId + ')');
+        }
+        this.__handleCompleted(data);
+      }
+    },
+    handleError : function(error, plugin) {
+      this.__log('Error on layer ' +  error.layerId + ':' + error.message);
+      var status = this.getLayerStatus(error.layerId);
+      status.error.push(error.message);
+      var requestBuilder = Oskari.requestBuilder('ShowMessageRequest');
+      var layer = this.getMapLayer(error.layerId);
 
-	});
+      if( this._errorLayer === null ) {
+        this._errorLayer = error;
+      } else {
+        for(var i = 0; i < this._errorLayers.length; i++) {
+          //same layer don't do anythin
+          if( this._errorLayers[i].errorlayer.layerId === error.layerId) {
+            return;
+          }
+        }
+      }
+      if( error.type === 'normal' && !error.success) {
+        if(requestBuilder){
+          var request = requestBuilder(layer._name+": could not load layer", 'error');
+          Oskari.getSandbox().request('system-message', request);
+        } else {
+          Oskari.log(this.getName(), "no system-message started");
```

Comment:
The syntax for log is: Oskari.log(this.getName()).info("no system-message started");